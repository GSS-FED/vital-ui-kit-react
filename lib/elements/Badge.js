// @flow

import * as React from 'react';
import styled from 'styled-components';

const SgBadge = styled.span`
  display: inline-block;
  vertical-align: middle;
  color: #ffffff;
  background-color: #00c3ff;
  padding: 0.25rem 0.75rem;
  border-radius: 0.75rem;
  font-size: 0.725rem;
`;

type Props = {
  label: string,
  children?: React.Node,
};

const Badge = ({ label, children, ...props }: Props) => (
  <SgBadge {...props}>
    {label}
    {children}
  </SgBadge>
);

export default Badge;
