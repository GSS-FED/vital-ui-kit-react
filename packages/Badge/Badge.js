/**
 * @flow
 * Copyright Â© 2017 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
 * MIT license
 */

import * as React from 'react';
import styled from 'styled-components';

const Root = styled.span`
  display: inline-block;
  vertical-align: middle;
  color: #ffffff;
  background-color: #00c3ff;
  padding: 0.25rem 0.75rem;
  border-radius: 0.75rem;
  font-size: 0.725rem;
`;

type Props = {
  /** Text on the badge */
  label: string | number,
  children?: React.Node,
};

/**
 * @render react
 * @name Badge
 * @description Vital UI Kit Badge Label
 * @example
 * <Badge label="99+" />
 */

const Badge = ({ label, children, ...props }: Props) => (
  <Root {...props}>
    {label}
    {children}
  </Root>
);

export default Badge;
