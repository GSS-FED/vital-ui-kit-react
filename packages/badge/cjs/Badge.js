'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _templateObject = _taggedTemplateLiteral(['\n  display: inline-block;\n  vertical-align: middle;\n  padding: 0.25rem 0.75rem;\n  border-radius: 0.75rem;\n  font-size: 0.725rem;\n\n  ', ';\n'], ['\n  display: inline-block;\n  vertical-align: middle;\n  padding: 0.25rem 0.75rem;\n  border-radius: 0.75rem;\n  font-size: 0.725rem;\n\n  ', ';\n']);

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _reactTheme = require('@vital-ui/react-theme');

var _reactUtils = require('@vital-ui/react-utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); } /**
                                                                                                                                                   * 
                                                                                                                                                   * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
                                                                                                                                                   * MIT license
                                                                                                                                                   */

var inverseOrNot = function inverseOrNot(_ref) {
  var inverse = _ref.inverse,
      theme = _ref.theme;
  return '\n  background-color: ' + (inverse ? theme.badge.inverseBg : theme.badge.bg) + ';\n  color: ' + (inverse ? theme.badge.inverseColor : theme.badge.color) + ';\n';
};

var Root = _styledComponents2.default.span(_templateObject, inverseOrNot);

Root.defaultProps = {
  theme: _reactTheme.defaultTheme
};

/**
 * @render react
 * @name Badge
 * @description Vital UI Kit Badge Label
 * @example
 * <Badge label="99+" />
 */

var Badge = function Badge(_ref2) {
  var label = _ref2.label,
      trunc = _ref2.trunc,
      inverse = _ref2.inverse,
      props = _objectWithoutProperties(_ref2, ['label', 'trunc', 'inverse']);

  return React.createElement(
    Root,
    _extends({ inverse: inverse }, props),
    trunc ? (0, _reactUtils.trunTo)(label) : label
  );
};

Badge.defaultProps = {
  trunc: true,
  inverse: false
};

exports.default = Badge;