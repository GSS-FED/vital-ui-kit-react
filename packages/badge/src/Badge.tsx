/**
 * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
 * MIT license
 */

import * as React from 'react';
import styled from 'styled-components';
import { defaultTheme } from '@vital-ui/react-theme';
import {
  trunTo,
  superBoxStyle,
  CssProps,
  css,
} from '@vital-ui/react-utils';
import cn from 'classnames';

const inverseOrNot = ({
  inverse,
  theme,
}: {
  inverse: boolean;
  theme: any;
}) => `
  background-color: ${
    inverse ? theme.badge.inverseBg : theme.badge.bg
  };
  color: ${inverse ? theme.badge.inverseColor : theme.badge.color};
`;

interface RootProps extends CssProps {
  inverse: boolean;
}

const Root = styled<RootProps, 'span'>('span')`
  display: inline-block;
  vertical-align: middle;
  padding: 0.25rem 0.75rem;
  border-radius: 0.75rem;
  font-size: 0.725rem;
  ${superBoxStyle};
  ${inverseOrNot};
  ${css};
`;

Root.defaultProps = {
  theme: defaultTheme,
};

export interface BadgeProps extends CssProps {
  /** Text on the badge */
  label: string | number;
  /** Show 99+ if number is more than 100 */
  trunc?: boolean;
  /** nIverse style */
  inverse?: boolean;
  /** className, default is 'vital__badge' */
  className?: string;
  style?: React.CSSProperties;
}

/**
 * @render react
 * @name Badge
 * @description Vital UI Kit Badge Label
 * @example
 * <Badge label="99+" />
 */

export const Badge: React.SFC<BadgeProps> = ({
  label,
  trunc = true,
  inverse = false,
  className,
  style,
  ...props
}) => (
  <Root
    inverse={inverse}
    className={cn('vital__badge', className)}
    style={style}
    {...props}
  >
    {trunc ? trunTo(label) : label}
  </Root>
);
