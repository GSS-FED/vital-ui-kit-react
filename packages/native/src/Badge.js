// @flow
/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import styled from 'styled-components/native';

const Root = styled.View`
  padding-left: 7;
  padding-right: 7;
  background-color: ${props => props.bg || '#D8D8D8'};
  border-radius: 7.5;
  height: 16;
`;

const Label = styled.Text`
  color: ${props => props.color || '#3D414A'};
  font-size: 12;
`;

export type BadgeProps = {
  label: string | number,
  bg?: string,
  color?: string,
};

const Badge = ({ label, bg, color, ...props }: BadgeProps) => (
  <Root bg={bg} {...props}>
    <Label color={color}>{label}</Label>
  </Root>
);

export default Badge;
