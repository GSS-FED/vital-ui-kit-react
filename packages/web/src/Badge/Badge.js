/**
 * @flow
 * Copyright Â© 2017 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
 * MIT license
 */

import * as React from 'react';
import styled from 'styled-components';

import { trunTo } from '../utils/math';

const inverseStyle = ({ inverse }) => inverse &&`
  background-color: #ffffff;
  color: #00c3ff;
`

const Root = styled.span`
  display: inline-block;
  vertical-align: middle;
  color: #ffffff;
  background-color: #00c3ff;
  padding: 0.25rem 0.75rem;
  border-radius: 0.75rem;
  font-size: 0.725rem;

  ${inverseStyle}
`;

type Props = {
  /** Text on the badge */
  label: string | number,
  /** show 99+ if number is more than 100 */
  trunc?: boolean,
  inverse?: boolean,
};

/**
 * @render react
 * @name Badge
 * @description Vital UI Kit Badge Label
 * @example
 * <Badge label="99+" />
 */

const Badge = ({ label, trunc = true, inverse = false, ...props }: Props) => (
  <Root inverse={inverse} {...props}>
    {trunc ? trunTo(label) : label}
  </Root>
);

export default Badge;
