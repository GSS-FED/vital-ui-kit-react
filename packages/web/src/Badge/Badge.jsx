/**
 * @flow
 * Copyright Â© 2018 Galaxy Software Services https://github.com/GSS-FED/vital-ui-kit-react
 * MIT license
 */

import * as React from 'react';
import styled from 'styled-components';

import { trunTo } from '../utils/math';

const inverseOrNot = ({ inverse, theme }) => `
  background-color: ${inverse ? theme.badge.inverseBg : theme.bg};
  color: ${inverse ? theme.inverseColor : theme.color};
`;

const Root = styled.span`
  display: inline-block;
  vertical-align: middle;
  padding: 0.25rem 0.75rem;
  border-radius: 0.75rem;
  font-size: 0.725rem;

  ${inverseOrNot};
`;

type Props = {
  /** Text on the badge */
  label: string | number,
  /** Show 99+ if number is more than 100 */
  trunc?: boolean,
  /** nIverse style */
  inverse?: boolean,
};

/**
 * @render react
 * @name Badge
 * @description Vital UI Kit Badge Label
 * @example
 * <Badge label="99+" />
 */

const Badge = ({ label, trunc, inverse, ...props }: Props) => (
  <Root inverse={inverse} {...props}>
    {trunc ? trunTo(label) : label}
  </Root>
);

Badge.defaultProps = {
  trunc: true,
  inverse: false
};

export default Badge;
