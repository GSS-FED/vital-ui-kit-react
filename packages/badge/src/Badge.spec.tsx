import React from 'react';
import rerender from 'react-test-renderer';
import { Badge } from './Badge';

describe('Badge', () => {
  it('should render with default styles', () => {
    const badge = rerender.create(<Badge>1</Badge>);
    expect(badge.toJSON()).toMatchSnapshot();
  });
});
