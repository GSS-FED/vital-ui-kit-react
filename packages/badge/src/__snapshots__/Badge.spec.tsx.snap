// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render with default styles 1`] = `
.c0 {
  display: inline-block;
  vertical-align: middle;
  padding: 0.25rem 0.75rem;
  border-radius: 0.75rem;
  font-size: 0.725rem;
  background-color: #00C3FF;
  color: #ffffff;
}

<span
  className="vital__badge c0"
>
  1
</span>
`;
